.github/workflows/_openxla_pin_update_weekly.yml

```diff
@@ -0,0 +1,52 @@
+name: Weekly Code Update PR
+on:
+  schedule:
+    - cron: '0 9 * * 1'  # Each Monday 9:00AM (PDT time)
+
+jobs:
+  update_and_create_pr:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get current date
+        run: echo "CURRENT_DATE=$(date +'%Y%m%d')" >> $GITHUB_ENV
+
+      - name: Install sed
+        run: sudo apt-get update && sudo apt-get install -y sed
+
+      - name: Update date in setup.py
+        run: |
+          sed -i "s/_date =.*/_date = '${{ env.CURRENT_DATE }}'/" ${FILE_PATH}
+          git add ${FILE_PATH}
+          git commit -m "Weekly update setup.py to (${{ env.CURRENT_DATE }})"
+          git push origin HEAD:weekly-update
+        env:
+          FILE_PATH: setup.py
+
+      - name: Fetch HEAD commit from OpenXLA github repo
+        run: |
+          git remote add target https://github.com/openxla/xla.git
+          git fetch target
+          echo "HEAD_COMMIT=$(git rev-parse target/main)" >> $GITHUB_ENV
+
+      - name: Update OpenXLA commit in WORKSPACE
+        run: |
+          sed -i "s/xla_hash =.*/xla_hash = '${{ env.GITHUB_ENV }}'/" ${FILE_PATH}
+          git add ${FILE_PATH}
+          git commit -m "Weekly update WORKSPACE with (${{ env.GITHUB_ENV }})"
+          git push origin HEAD:weekly-update
+        env:
+          FILE_PATH: WORKSPACE
+
+      - name: Create Pull Request
+        uses: actions/create-pull-request@v4
+        with:
```

Comment:
Is there a way to tag this with the `tpuci` label?

Reply:
Thanks, updated